conf/routines.sql

```diff
@@ -2180,24 +2238,56 @@ DELIMITER ;
 /*!50003 SET character_set_results = utf8 */ ;
 /*!50003 SET collation_connection  = utf8_general_ci */ ;
 /*!50003 SET @saved_sql_mode       = @@sql_mode */ ;
-/*!50003 SET sql_mode              = '' */ ;
+/*!50003 SET sql_mode              = 'STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION' */ ;
 DELIMITER ;;
 CREATE PROCEDURE `retrieve_current_sessions_for_person`(IN p_beamline varchar(15), IN p_fed_id varchar(24), IN p_tolerance_minutes int)
+    READS SQL DATA
+    COMMENT 'Returns a multi-row result-set with the current (within tolerance p_tolerance_minutes)\nsession(s) (mx12345-123), their start and end dates for person p_fed_id and beamline p_beamline'
 BEGIN
-	set p_tolerance_minutes = IFNULL(p_tolerance_minutes, 0);
+	  SET p_tolerance_minutes = IFNULL(p_tolerance_minutes, 0);
     SELECT concat(p.proposalCode, p.proposalNumber, '-', bs.visit_number) `session`, bs.startDate, bs.endDate
     FROM Proposal p
       INNER JOIN BLSession bs on p.proposalId = bs.proposalId
-      INNER JOIN Session_has_Person shp ON shp.sessionId = bs.sessionId 
-	  INNER JOIN Person per ON shp.personId = per.personId
-	WHERE bs.beamlinename = p_beamline AND per.login = p_fed_id AND now() BETWEEN bs.startDate - INTERVAL p_tolerance_minutes MINUTE and bs.endDate + INTERVAL p_tolerance_minutes MINUTE
+      INNER JOIN Session_has_Person shp ON shp.sessionId = bs.sessionId
+	    INNER JOIN Person per ON shp.personId = per.personId
+	  WHERE bs.beamlinename = p_beamline AND bs.visit_number <> 0 AND per.login = p_fed_id AND now() BETWEEN bs.startDate - INTERVAL p_tolerance_minutes MINUTE and bs.endDate + INTERVAL p_tolerance_minutes MINUTE
     ORDER BY bs.startDate;
 END ;;
 DELIMITER ;
 /*!50003 SET sql_mode              = @saved_sql_mode */ ;
 /*!50003 SET character_set_client  = @saved_cs_client */ ;
 /*!50003 SET character_set_results = @saved_cs_results */ ;
 /*!50003 SET collation_connection  = @saved_col_connection */ ;
+/*!50003 DROP PROCEDURE IF EXISTS `retrieve_dc_group` */;
+/*!50003 SET @saved_cs_client      = @@character_set_client */ ;
+/*!50003 SET @saved_cs_results     = @@character_set_results */ ;
+/*!50003 SET @saved_col_connection = @@collation_connection */ ;
+/*!50003 SET character_set_client  = utf8 */ ;
+/*!50003 SET character_set_results = utf8 */ ;
+/*!50003 SET collation_connection  = utf8_general_ci */ ;
+/*!50003 SET @saved_sql_mode       = @@sql_mode */ ;
+/*!50003 SET sql_mode              = 'STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION' */ ;
+DELIMITER ;;
+CREATE PROCEDURE `retrieve_dc_group`(p_id int unsigned)
+    READS SQL DATA
+    COMMENT 'Returns a single-row result-set with the columns for the given data collection group id'
+BEGIN
+    IF p_id IS NOT NULL THEN
+      SELECT sessionId, blSampleId "sampleId", experimentType "experimenttype", startTime "starttime", endTime "endtime",
```

Comment:
Question on this - why rewrite `experimentType`, `startTime` and `endTime` with lower case?

Reply:
I think the reason is that the Java API already has a "bean" for DataCollectionGroup (already used by the upsert methods), and this unfortunately has these fields/attributes in lowercase. Silly, I know.